### DISCUSSION TITLE
Follow/Open url in floating window

### DISCUSSION THREAD

**Discussion by NoahTheDuke** [neoclide/coc.nvim#3123] [Q&A] [answered]
I'd love to be able to open a url embedded in the docs of a given object:
<img width="740" alt="image" src="https://user-images.githubusercontent.com/603677/120703544-72294800-c483-11eb-877e-12bec5db232f.png">
Is this possible?

**Comment by chemzqm** [comment, marked as answer]
My key-mappings to open url:
``` vim
nnoremap <silent> <leader>o :call <SID>Open()<CR>
function! s:Open()
  let res = CocAction('openLink')
  if res | return | endif
  let line = getline('.')
  " match url
  let url = matchstr(line, '\vhttps?:\/\/[^)\]''" ]+')
  if !empty(url)
    let output = system('open '. url)
  else
    let mail = matchstr(line, '\v([A-Za-z0-9_\.-]+)\@([A-Za-z0-9_\.-]+)\.([a-z\.]+)')
    if !empty(mail)
      let output = system('open mailto:' . mail)
    else
      let output = system('open ' . expand('%:p:h'))
    endif
  endif
  if v:shell_error && output !=# ""
    echoerr output
  endif
endfunction
```
you can jump to float window of neovim, but not popup of vim.

**Reply by NoahTheDuke** [reply]
This is a cool function! Not quite what I'm asking for tho. Is it possible to access the link in the floating window?

**Comment by jackielii** [comment]
I use https://github.com/tyru/open-browser.vim but @chemzqm 's solution is better: it directly opens the link if the symbol under the cursor has a doc!

**Comment by arthur-zhuk** [comment]
do you mean like hovering over the URL and pressing `gx`?
